webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -245,28 +254,39 @@ public void onNext(DataChunk data) {
             throw new IllegalStateException("Response is already closed!");
         }
         if (data != null) {
-
-            LOGGER.finest(() -> log("Sending data chunk"));
-
             DefaultHttpContent httpContent = new DefaultHttpContent(Unpooled.wrappedBuffer(data.data()));
-
             LOGGER.finest(() -> log("Sending data chunk on event loop thread."));
 
             ChannelFuture channelFuture;
+            chunksInFlight.incrementAndGet();
             if (data.flush()) {
                 channelFuture = ctx.writeAndFlush(httpContent);
             } else {
                 channelFuture = ctx.write(httpContent);
             }
-
             channelFuture
                     .addListener(future -> {
                         data.release();
                         LOGGER.finest(() -> log("Data chunk sent with result: " + future.isSuccess()));
+                        if (future.isSuccess()) {
+                            chunksInFlight.decrementAndGet();
+                            chunksWritten.incrementAndGet();
```

Comment:
An uninformed, naïve question: do these by any chance need to be mutated atomically together?